packages/node/src/runtimePlugin.ts

```diff
@@ -0,0 +1,380 @@
+  };
+  // no baseURI
+
+  // object to store loaded chunks
+  // "0" means "already loaded", Promise means loading
+  var installedChunks = {};
+
+  // no on chunks loaded
+
+  var installChunk = __webpack_require__.C
+    ? __webpack_require__.C
+    : (chunk) => {
+        var moreModules = chunk.modules,
+          chunkIds = chunk.ids,
+          runtime = chunk.runtime;
+        for (var moduleId in moreModules) {
+          if (__webpack_require__.o(moreModules, moduleId)) {
+            __webpack_require__.m[moduleId] = moreModules[moduleId];
+          }
+        }
+        if (runtime) runtime(__webpack_require__);
+        for (var i = 0; i < chunkIds.length; i++) {
+          if (installedChunks[chunkIds[i]]) {
+            installedChunks[chunkIds[i]][0]();
+          }
+          installedChunks[chunkIds[i]] = 0;
+        }
+      };
+  // load script equivalent for server side
+  __webpack_require__.l = function (url, callback, chunkId) {
+    console.log('LOADING CONTAINER', url);
+    if (!chunkId) {
+      throw new Error('__webpack_require__.l name is required for ' + url);
+    }
+    var usesInternalRef = chunkId.indexOf('__webpack_require__') === 0;
+    if (usesInternalRef) {
+      var regex =
+        /__webpack_require__\.federation\.instance\.moduleCache\.get\(([^)]+)\)/;
+      var match = chunkId.match(regex);
+      if (match) {
+        chunkId = match[1].replace(/["']/g, '');
+      }
+    }
+    __webpack_require__.federation.runtime
+      .loadScriptNode(url, { attrs: {} })
+      .then(function (res) {
+        var federation = __webpack_require__.federation;
+        var enhancedRemote = federation.instance.initRawContainer(
+          chunkId,
+          url,
+          res,
+        );
+        // use normal global assignment
+        if (!usesInternalRef && !globalThis[chunkId]) {
+          globalThis[chunkId] = enhancedRemote;
+        }
+        console.log('adding remote', chunkId);
+        callback(enhancedRemote);
+      })
+      .catch(function (error) {
+        callback(error);
+      });
+  };
+  // Dynamic filesystem chunk loading for javascript
+  __webpack_require__.f.readFileVm = function (chunkId, promises) {
+    var installedChunkData = installedChunks[chunkId];
+    if (installedChunkData !== 0) {
+      // 0 means "already installed".
+      // array of [resolve, reject, promise] means "currently loading"
+      if (installedChunkData) {
+        promises.push(installedChunkData[2]);
+      } else {
+        console.log(
+          'is outbound chunk handler ref',
+          __webpack_require__.federation.chunkMatcher(chunkId),
+        );
+        // console.log(__webpack_require__.federation.bundlerRuntimeOptions);
+        // console.log(__webpack_require__.federation.initOptions.name);
+        if (
+          // check if real chunk for handler. Federation makes virtual chunks that are handled by other handlers
+          // __webpack_require__.federation.chunkMatcher(chunkId)
```

Comment:
as an alternative to calling mock chunk callback, i can take the logic that commonjs plugin uses to determine chunk handler logic and expose it to our runtime. Then we can also check if this chunk should be processed by readFileVm, not .f.remotes, or .f.shared -> which later will call .f.readFileVm if no share module exists already. 

Each runtime chunk provides different logic for matching its handlers.

Reply:
https://github.com/module-federation/universe/pull/2013/files#diff-6e1ea38caf351843e18282c10b22f061d98f2d0a6a17a4dde25d9162c0d6ca3aR143